crates/core/tedge_api/src/entity_store.rs

```diff
@@ -161,12 +165,12 @@ impl EntityStore {
             affected_entities.push(parent.clone());
         }
 
-        let external_id = message
-            .external_id
-            .unwrap_or_else(|| self.derive_external_id(&message.mqtt_id));
+        let entity_id = message
+            .entity_id
+            .unwrap_or_else(|| self.derive_entity_id(&message.mqtt_id));
```

Comment:
This change is really good one, as external id has a specific meaning in the context of c8y.